packages/clay-autocomplete/src/LoadingIndicator.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import Context from './Context';
+import React, {useContext, useEffect} from 'react';
+
+const LoadingIndicatorMarkup: React.FunctionComponent<
+	React.HTMLAttributes<HTMLDivElement>
+> = ({children, className, ...otherProps}) => (
+	<div
+		{...otherProps}
+		className={classNames(
+			'input-group-inset-item input-group-inset-item-after',
+			className
+		)}
+	>
+		<span className="inline-item inline-item-middle">{children}</span>
+	</div>
+);
+
+export interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Div component to render. It can be a one component that will replace the markup.
+	 */
+	component?:
+		| typeof LoadingIndicatorMarkup
```

Comment:
Is it necessary to define this here if we already declare `React.ComponentType<any>`?

Reply:
Yes, I think maybe not, I've seen this double statement in some codebase but I can not understand the real reason, if it's just enforcing the use of `LoadingIndicatorMarkup` for example. I'll remove that.